samples/MicStream.js

```diff
@@ -0,0 +1,378 @@
+    .start({
+      sampleRateHertz: sampleRateHertz,
+      threshold: 0, //silence threshold
+      // Other options, see https://www.npmjs.com/package/node-record-lpcm16#options
+      //verbose: true, //CF: verbose true just outputs 1/2 the samplerateHertz every 1/2 sec.  Not all that necessary.
+      recordProgram: 'rec', // was 'rec' : Try also "arecord" or "sox"
+      silence: '5.0', //seconds of silence before ending
+      //device: null
+    })
+    .on(`data`, chunk => {
+      file.write(chunk);
+    })
+    .on('error', console.error)
+    .pipe(recognizeStream);
+
+  console.log(
+    `micStreamRecognize: Listening for ${secondsofRecording} seconds, press Ctrl+C to stop.`
+  );
+
+  //CF: records for secondsofRecording seconds.
+  setTimeout(function() {
+    record.stop();
+    console.log(`micStreamRecognize: Recording complete!`);
+    console.log(
+      `micStreamRecognize: There are ${
+        InterimResults.length
+      } InterimResults generated from the recording`
+    );
+    console.log(
+      `**Note that best guess/ non-interim results below are included below, true, and don't include confidence value**`
+    );
+
+    for (var i in InterimResults) {
+      console.log(`  `, i, `)`, InterimResults[i].toString());
+    }
+    eventEmitter.emit(`micStreamRecognize: complete`);
+  }, secondsofRecording * 1000);
+  return InterimResults;
+  // [END micStreamRecognize]
+}
+
+function wavFileToText(filename, encoding, sampleRateHertz, languageCode) {
+  // [START wavFileToText]
+
+  const config = {
+    encoding: encoding,
+    sampleRateHertz: sampleRateHertz,
+    languageCode: languageCode,
+  };
+  eventEmitter.emit(`test_1: Started`);
+  eventEmitter.emit(`wavFileToText: Started`);
+  // Imports the Google Cloud client library
+  const speech = require('@google-cloud/speech');
+
+  // Imports fs to save recording locally
+  const fs = require('fs');
+
+  // Creates a client
+  const client = new speech.SpeechClient();
+
+  const path = require(`path`);
+
+  filename = path.resolve(filename);
+  console.log(`resolved filename ${filename}`);
+  const audio = {
+    content: fs.readFileSync(filename).toString('base64'),
+  };
+
+  const request = {
+    config: config,
+    audio: audio,
+  };
+
+  // Detects speech in the audio file
+  client
+    .recognize(request)
+    .then(data => {
+      const response = data[0];
+      const transcription = response.results
+        .map(result => result.alternatives[0].transcript)
+        .join('\n');
+      console.log(`wavFileToText: Transcription; \n`, transcription);
+      wavtranscription = transcription;
+      const confidence = response.results
+        .map(result => result.alternatives[0].confidence)
+        .join('\n');
+      console.log(`wavFileToText: Confidence; \n`, confidence);
+      eventEmitter.emit(`wavFileToText: complete`);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END wavFileToText]
+}
+
+function test_1(
```

Comment:
Please move tests out of this file & into the system-tests

e.g. how about this:

`microphoneStreaming.js` & `system-tests/microphoneStreaming.test.js`